### PR TITLE
ON-36997 # Added auto focusing of first element on page when tabbing …

### PR DESCRIPTION
**Pull request by Zaxist**
…forwards

### CODE REVIEW

File: src/components/renderer/PageFormElements.tsx [hunk 1]
```diff
@@ -37,6 +37,28 @@ function PageFormElements({
...
   setFormSubmission,
 }: Props) {
+  // Effect responsible for focusing the first element on a page when the next button is focused and used to nav forwards a page
+  React.useEffect(() => {
+    if (isActive) {
+      console.log(`${pageElement.label} page became active`)
+      const currentFocusedElement = document.activeElement
+      if (currentFocusedElement?.getAttribute('data-nav-button') === 'next') {
+        const selector =
+          'a:not([disabled]), button:not([disabled]), input:not([disabled]), select:not([disabled]), [tabindex]:not([disabled]):not([tabindex="-1"])'
```
Comment (divporter) [inline src/components/renderer/PageFormElements.tsx:47]: does *:not([disabled]) work?

**Review by divporter — APPROVED** [review]

File: src/components/renderer/PageFormElements.tsx (hunk 1 above)
Comment (Zaxist) [reply src/components/renderer/PageFormElements.tsx]: That will give us things that we can't focus.

File: src/components/renderer/PageFormElements.tsx [hunk 2]
```diff
@@ -37,6 +37,28 @@ function PageFormElements({
...
+      if (currentFocusedElement?.getAttribute('data-nav-button') === 'next') {
+        const selector =
+          'a:not([disabled]), button:not([disabled]), input:not([disabled], [hidden]), textarea:not([disabled]), select:not([disabled]), [tabindex]:not([disabled]):not([tabindex="-1"])'
+        const focusableElementsOnPage = document
+          .getElementById(pageElement.id)
+          ?.querySelectorAll(selector)
+        if (focusableElementsOnPage?.length) {
+          const firstElementOnPage = focusableElementsOnPage[0]
+          if (firstElementOnPage) {
+            // @ts-expect-error Does have a focus method
```
Comment (mymattcarroll) [inline src/components/renderer/PageFormElements.tsx:54]: Can we avoid this by using `instanceof`?

File: src/components/renderer/PageFormElements.tsx [hunk 3]
```diff
@@ -37,6 +37,28 @@ function PageFormElements({
...
+      const currentFocusedElement = document.activeElement
+      if (currentFocusedElement?.getAttribute('data-nav-button') === 'next') {
+        const selector =
+          'a:not([disabled]), button:not([disabled]), input:not([disabled], [hidden]), textarea:not([disabled]), select:not([disabled]), [tabindex]:not([disabled]):not([tabindex="-1"])'
+        const focusableElementsOnPage = document
+          .getElementById(pageElement.id)
+          ?.querySelectorAll(selector)
+        if (focusableElementsOnPage?.length) {
+          const firstElementOnPage = focusableElementsOnPage[0]
+          if (firstElementOnPage) {
```
Comment (mymattcarroll) [inline src/components/renderer/PageFormElements.tsx:53]:
```suggestion
          if (firstElementOnPage instanceof HTMLElement) {
```

File: src/components/renderer/PageFormElements.tsx (hunk 2 above)
Comment (mymattcarroll) [inline src/components/renderer/PageFormElements.tsx:54]:
```suggestion
          if (firstElementOnPage instanceof HTMLElement) {
```

**Pull request merged**